packages/web/src/common/store/upload/sagas.ts

```diff
@@ -104,8 +105,20 @@ function* combineMetadata(
     metadata.tags = collectionMetadata.trackDetails.tags
   }
 
-  if (collectionMetadata.is_downloadable) {
-    metadata.is_downloadable = true
+  // Set download & hidden status
+  metadata.is_downloadable = !!collectionMetadata.is_downloadable
+
+  metadata.is_unlisted = !!collectionMetadata.is_unlisted
+  if (collectionMetadata.is_unlisted && collectionMetadata.field_visibility) {
+    // Convert any undefined values to booleans
+    const booleanFieldArray = Object.entries(
+      collectionMetadata.field_visibility
+    ).map(([k, v]) => [k, !!v]) as [string, boolean][]
+    const booleanFieldVisibility = booleanFieldArray.reduce(
+      (acc, [k, v]) => ({ [k]: v, ...acc }),
+      {}
```

Comment:
FYI: [_.fromPairs](https://lodash.com/docs/4.17.15#fromPairs) will also do this part

Also, if I understand this correctly, this might just do the whole thing:
```metadata.field_visibility = _.mapValues(collectionMetadata.field_visibility, Boolean)```